crates/framework/src/natives/mos_stdlib/object_extension/mod.rs

```diff
@@ -0,0 +1,174 @@
+// Copyright (c) RoochNetwork
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use better_any::{Tid, TidAble};
+use mos_types::object::{Object, ObjectID, Owner};
+use move_binary_format::errors::PartialVMError;
+use move_core_types::{
+    account_address::AccountAddress, language_storage::StructTag, value::MoveStructLayout,
+    vm_status::StatusCode,
+};
+use move_vm_types::{
+    loaded_data::runtime_types::Type,
+    values::{Struct, Value},
+};
+use std::{
+    cell::RefCell,
+    collections::{BTreeMap, BTreeSet},
+};
+
+/// A table resolver which needs to be provided by the environment. This allows to lookup
+/// data in remote storage, as well as retrieve cost of table operations.
+pub trait ObjectResolver {
+    fn resolve_object(&self, object_id: ObjectID) -> Result<Option<Object>, anyhow::Error>;
+
+    // fn delete_object(&self,
+    //     object_id: ObjectID,
+    // ) -> Result<(), anyhow::Error>;
+}
+
+pub enum TransferResult {
+    New,
+    SameOwner,
+    OwnerChanged,
+}
+/// The native object context extension. This needs to be attached to the NativeContextExtensions
+/// value which is passed into session functions, so its accessible from natives of this
+/// extension.
+#[derive(Tid)]
+pub struct NativeObjectContext<'a> {
+    resolver: &'a dyn ObjectResolver,
+    object_data: RefCell<ObjectData>,
+}
+
+impl<'a> NativeObjectContext<'a> {
+    pub fn new(resolver: &'a dyn ObjectResolver) -> Self {
+        Self {
+            resolver,
+            object_data: RefCell::new(ObjectData::default()),
+        }
+    }
+
+    pub fn get_object(&self, object_id: ObjectID) -> Result<Option<Object>, PartialVMError> {
+        let object_data = self.object_data.borrow();
+        //TODO need to check the transfers
+        // if let Some(object) = object_data.transfers.get(&object_id) {
```

Comment:
Need to uncomment?

Reply:
Maybe we do not need to get objects from `transfers`, because the `get_object` is only used in the argument resolver now. if we provide some native function and allow the dev to get objects by id in the Move modules, we need to do this.  let it be TBD.